src/vulkan/vk-shader-object.cpp

```
@@ -343,46 +343,59 @@ Result BindingDataBuilder::bindAsEntryPoint(
     uint32_t entryPointIndex
 )
 {
-    BindingOffset offset = inOffset;
+    if (layout->getSlangLayout()->getStage() != SLANG_STAGE_RAY_GENERATION)
```

Cppcheck (2.21.0)

[portability] Shifting signed 32-bit value by 31 bits is implementation-defined behaviour

(shiftTooManyBitsSigned)